Source/Laser/LaserParticleContainer.cpp

```diff
@@ -119,19 +119,19 @@ LaserParticleContainer::LaserParticleContainer (AmrCore* amr_core, int ispecies,
     }
 
     // The first polarization vector
-    s = 1.0/std::sqrt(p_X[0]*p_X[0] + p_X[1]*p_X[1] + p_X[2]*p_X[2]);
+    s = 1.0_rt / std::sqrt(p_X[0]*p_X[0] + p_X[1]*p_X[1] + p_X[2]*p_X[2]);
     p_X = { p_X[0]*s, p_X[1]*s, p_X[2]*s };
 
-    Real dp = std::inner_product(nvec.begin(), nvec.end(), p_X.begin(), 0.0);
+    Real const dp = std::inner_product(nvec.begin(), nvec.end(), p_X.begin(), 0.0);
     AMREX_ALWAYS_ASSERT_WITH_MESSAGE(std::abs(dp) < 1.0e-14,
         "Laser plane vector is not perpendicular to the main polarization vector");
 
     p_Y = CrossProduct(nvec, p_X);   // The second polarization vector
 
-    s = 1.0/std::sqrt(stc_direction[0]*stc_direction[0] + stc_direction[1]*stc_direction[1] + stc_direction[2]*stc_direction[2]);
+    s = 1.0_rt / std::sqrt(stc_direction[0]*stc_direction[0] + stc_direction[1]*stc_direction[1] + stc_direction[2]*stc_direction[2]);
     stc_direction = { stc_direction[0]*s, stc_direction[1]*s, stc_direction[2]*s };
-    dp = std::inner_product(nvec.begin(), nvec.end(), stc_direction.begin(), 0.0);
-    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(std::abs(dp) < 1.0e-14,
+    Real const dp2 = std::inner_product(nvec.begin(), nvec.end(), stc_direction.begin(), 0.0);
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(std::abs(dp2) < 1.0e-14,
```

Comment:
for follow-ups: we want to use `std::numeric_limits<Real>::epsilon` for such numerical values.